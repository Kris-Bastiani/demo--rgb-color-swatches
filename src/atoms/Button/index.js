import React, { memo } from 'react';

const Button = ({ label, onClick }) => (<button onClick={onClick}>{label}</button>);

export default memo(Button);
