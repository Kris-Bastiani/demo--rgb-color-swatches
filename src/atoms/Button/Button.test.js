import React from 'react';
import { render } from '@testing-library/react';

import Button from '.';

describe('Component: Button', () => {
	it('should match snapshot when no props are passed', () => {
		const { container } = render(<Button />);
		expect(container).toMatchSnapshot();
	});

	it('should match snapshot when props are passed', () => {
		const { container } = render(<Button label='foobar' onClick={() => {}} />);
		expect(container).toMatchSnapshot();
	});
});
