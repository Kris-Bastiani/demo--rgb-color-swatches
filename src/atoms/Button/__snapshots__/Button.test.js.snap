// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: App should match snapshot when no props are passed 1`] = `
<div>
  <button />
</div>
`;

exports[`Component: App should match snapshot when props are passed 1`] = `
<div>
  <button>
    foobar
  </button>
</div>
`;
